### PR TITLE
Fixed keeping refs to class 'House' breaking init of unrelated classes

### PR DESCRIPTION
**Pull request by lava76**
This also fixes Expansion "simple territory" flag placing

### CODE REVIEW

**Comment by FlorianSW** [comment]
Tested on a local setup that I used to reproduce the territory flag issue:
- with the workshop release: create territory action broken
- with the built version of this PR: create territory action available

File: Scripts/3_Game/classes.c [hunk 1]
```diff
@@ -40,17 +40,23 @@ class _Vehicle {
...
-    void _Event(string className, string icon, Object _reference) {this.className = className; this.icon = icon; this._reference = _reference; this.id = EntityAI.Cast(this._reference).GetNetworkIDString(); }
+    //! Keeping object reference to class House (aka, wrecks from events) has weird side-effects for some reason,
+    //! it will break initialization of other, unrelated classes (EnfScript bug?).
+    //! Use network ID to get object dynamically when needed instead.
+    void _Event(string className, string icon, Object _reference) {this.className = className; this.icon = icon; _reference.GetNetworkID(id_low, id_high); this.id = id_high.ToString() + id_low.ToString(); }
     string GetID() { return this.id; }
     string Class() { return this.className; }
     string Icon() { return this.icon; }
-    Object Ref() { return this._reference; }
+    Object Ref() { return GetGame().GetObjectByNetworkId(id_low, id_high); }
```
Comment (philippj) [inline Scripts/3_Game/classes.c:55]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/3_Game/classes.c [hunk 2]
```diff
@@ -40,17 +40,23 @@ class _Vehicle {
...
     bool Equals(_Event other) {
-        if(this._reference == NULL) return false;
-        return this._reference.GetPosition() == other._reference.GetPosition();
+        Object _reference = Ref();
```
Comment (philippj) [inline Scripts/3_Game/classes.c:57]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!CarScript.c [hunk 3]
```diff
@@ -1,6 +1,7 @@
 modded class CarScript {
-    private ref _Vehicle _registeredInstance = new _Vehicle(this);
+    private ref _Vehicle _registeredInstance;
     void CarScript() {
+        _registeredInstance = new _Vehicle(this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!CarScript.c:4]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!DayZAnimal.c [hunk 4]
```diff
@@ -1,7 +1,8 @@
 modded class AnimalBase  {
-    private ref _AI _registeredInstance = new _AI(this, false);
+    private ref _AI _registeredInstance;
 
     void AnimalBase () {
+        _registeredInstance = new _AI(this, false);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!DayZAnimal.c:5]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!DayZInfected.c [hunk 5]
```diff
@@ -1,7 +1,8 @@
 modded class ZombieBase  {
-    private ref _AI _registeredInstance = new _AI(this, true);
+    private ref _AI _registeredInstance;
 
     void ZombieBase () {
+        _registeredInstance = new _AI(this, true);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!DayZInfected.c:5]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!HeliCrash.c [hunk 6]
```diff
@@ -1,8 +1,9 @@
 modded class Wreck_UH1Y {
-    ref _Event _registeredInstance = new _Event(this.GetType(), "helicopter", this);
+    ref _Event _registeredInstance;
 
     override void EEInit() {
         super.EEInit();
+        _registeredInstance = new _Event(this.GetType(), "helicopter", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!HeliCrash.c:6]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!HeliCrash.c [hunk 7]
```diff
@@ -17,10 +18,11 @@ modded class Wreck_UH1Y {
 };
 
 class Wreck_Mi8 extends House {
-        ref _Event _registeredInstance = new _Event(this.GetType(), "helicopter", this);
+        ref _Event _registeredInstance;
 
         override void EEInit() {
             super.EEInit();
+            _registeredInstance = new _Event(this.GetType(), "helicopter", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!HeliCrash.c:25]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!PoliceWreck.c [hunk 8]
```diff
@@ -1,8 +1,9 @@
 class Land_Wreck_hb01_aban1_police extends House {
-    ref _Event _registeredInstance = new _Event(this.GetType(), "car", this);
+    ref _Event _registeredInstance;
 
     override void EEInit() {
         super.EEInit();
+        _registeredInstance = new _Event(this.GetType(), "car", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!PoliceWreck.c:6]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!PoliceWreck.c [hunk 9]
```diff
@@ -37,10 +39,11 @@ class Land_Wreck_hb01_aban2_police extends House {
 };
 
 class Land_Wreck_sed01_aban1_police extends House {
-    ref _Event _registeredInstance = new _Event(this.GetType(), "car", this);
+    ref _Event _registeredInstance;
 
     override void EEInit() {
         super.EEInit();
+        _registeredInstance = new _Event(this.GetType(), "car", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!PoliceWreck.c:46]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!PoliceWreck.c [hunk 10]
```diff
@@ -18,10 +19,11 @@ class Land_Wreck_hb01_aban1_police extends House {
 };
 
 class Land_Wreck_hb01_aban2_police extends House {
-    ref _Event _registeredInstance = new _Event(this.GetType(), "car", this);
+    ref _Event _registeredInstance;
 
     override void EEInit() {
         super.EEInit();
+        _registeredInstance = new _Event(this.GetType(), "car", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!PoliceWreck.c:26]: Class attributes shall be accessed via `this.<attribute>`

File: Scripts/4_World/4_Hooking/!PoliceWreck.c [hunk 11]
```diff
@@ -56,10 +59,11 @@ class Land_Wreck_sed01_aban1_police extends House {
 };
 
 class Land_Wreck_sed01_aban2_police extends House {
-    ref _Event _registeredInstance = new _Event(this.GetType(), "car", this);
+    ref _Event _registeredInstance;
 
     override void EEInit() {
         super.EEInit();
+        _registeredInstance = new _Event(this.GetType(), "car", this);
```
Comment (philippj) [inline Scripts/4_World/4_Hooking/!PoliceWreck.c:66]: Class attributes shall be accessed via `this.<attribute>`

**Review by philippj — CHANGES_REQUESTED** [review]

**Comment by philippj** [comment]
Reverted in #10 .

`GetObjectByNetworkId` causes a heavy performance impact and is not feasible in production. Also, `GetPosition` on objects returned this way always give `0,0,0`.

**Pull request merged**